### PR TITLE
[core] Fixed static variable usage in the crypto module.

### LINKED ISSUE #3035: [BUG] Static Global State Variables Used for Multiple Connections
If you use the SRT library with multiple SRT connections in it. The global static state variables in srt/haicrypt/hcrypt_xpt_srt.c are used by all encrypted SRT connections. For instance if you have an application that is inputing and encrypted SRT stream and outputing an encrypted SRT stream. The following global static state variables are used by all of them at the same time:

```
srt/haicrypt/hcrypt_xpt_srt.c :

// Line: 68
static hcrypt_MsgInfo _hcMsg_SRT_MsgInfo;
// Line: 156
static hcrypt_MsgInfo _hcMsg_SRT_MsgInfo;
```

This can be observed with the srt-live-transmit application where you ijnput an ecrypted SRT source stream and output and encrypted SRT stream.

### PR DESCRIPTION
**Pull request by maxsharabayko**
As reported in #3035, `srt/haicrypt/hcrypt_xpt_srt.c` has two definitions of the static variable `_hcMsg_SRT_MsgInfo`:

```c
// Line: 68
static hcrypt_MsgInfo _hcMsg_SRT_MsgInfo;
// Line: 156
static hcrypt_MsgInfo _hcMsg_SRT_MsgInfo;
```

One more problem is the `.parseMsg` field off the structure that is initialized with a pointer to `hcryptMsg_SRT_ParseMsg` function. The `hcryptMsg_SRT_ParseMsg` function in its turn uses the static _hcMsg_SRT_MsgInfo internally.

To resolve this interdependency the `hcryptMsg_SRT_ParseMsg` function is made visible and now requires a pointer to a `hcrypt_MsgInfo`  structure instead of using the static variable implicitly.
The `_hcMsg_SRT_MsgInfo` can now be const and initialized once the following way:

```c
static const hcrypt_MsgInfo _hcMsg_SRT_MsgInfo = {
	.hdr_len     = HCRYPT_MSG_SRT_HDR_SZ,
	.pfx_len     = HCRYPT_MSG_SRT_PFX_SZ,
	.getKeyFlags = hcryptMsg_SRT_GetKeyFlags,
	.getPki      = hcryptMsg_SRT_GetPki,
	.setPki      = hcryptMsg_SRT_SetPki,
	.resetCache  = hcryptMsg_SRT_ResetCache,
	.indexMsg    = hcryptMsg_SRT_IndexMsg,
};
```

Fixes #3035.

### CODE REVIEW

File: haicrypt/hcrypt_xpt_srt.c [hunk 1]
```diff
@@ -110,7 +109,22 @@ static void hcryptMsg_SRT_IndexMsg(unsigned char *msg, unsigned char *pfx_cache)
...
+	.hdr_len     = HCRYPT_MSG_SRT_HDR_SZ,
+	.pfx_len     = HCRYPT_MSG_SRT_PFX_SZ,
+	.getKeyFlags = hcryptMsg_SRT_GetKeyFlags,
+	.getPki      = hcryptMsg_SRT_GetPki,
+	.setPki      = hcryptMsg_SRT_SetPki,
+	.resetCache  = hcryptMsg_SRT_ResetCache,
+	.indexMsg    = hcryptMsg_SRT_IndexMsg
+};
+
+const hcrypt_MsgInfo* hcryptMsg_SRT_InitMsgInfo(void)
```
Comment (ethouris) [inline haicrypt/hcrypt_xpt_srt.c:122]: But this "Init" function doesn't initialize anything. Just gets the pointer to the static global variable. Shouldn't be rather "Get" or something?

File: haicrypt/hcrypt_msg.h [hunk 2]
```diff
@@ -72,6 +71,7 @@ typedef struct {
 
 #define hcryptMsg_PaddedLen(len, fact)  ((((len)+(fact)-1)/(fact))*(fact))
 
+int hcryptMsg_SRT_ParseMsg(unsigned char* msg);
```
Comment (ethouris) [inline haicrypt/hcrypt_msg.h:74]: Well, why only this one? If it is ok to have this one function exported, why not all others as well?

And if this generic form of the solution was legit, maybe this function should remain "virtual"?

File: haicrypt/hcrypt_xpt_srt.c (hunk 1 above)
Comment (maxsharabayko) [reply haicrypt/hcrypt_xpt_srt.c]: Sorry, I rushed with the PR. It is just the same `hcryptMsg_SRT_MsgInfo` function that returns a pointer to the static variable.

File: haicrypt/hcrypt_msg.h (hunk 2 above)
Comment (maxsharabayko) [reply haicrypt/hcrypt_msg.h:74]: To minimize changes. I am not eager about rewriting the crypto module now 🙂  Only need to fix the bug.

Comment (maxsharabayko) [reply haicrypt/hcrypt_msg.h:74]: hcryptMsg_SRT_ParseMsg uses a pointer to `_hcMsg_SRT_MsgInfo`, while `_hcMsg_SRT_MsgInfo` has a pointer `.parseMsg    = hcryptMsg_SRT_ParseMsg`.

```c
static const hcrypt_MsgInfo _hcMsg_SRT_MsgInfo = {
	.hdr_len     = HCRYPT_MSG_SRT_HDR_SZ,
	.pfx_len     = HCRYPT_MSG_SRT_PFX_SZ,
	.getKeyFlags = hcryptMsg_SRT_GetKeyFlags,
	.getPki      = hcryptMsg_SRT_GetPki,
	.setPki      = hcryptMsg_SRT_SetPki,
	.resetCache  = hcryptMsg_SRT_ResetCache,
	.indexMsg    = hcryptMsg_SRT_IndexMsg,
	.parseMsg    = hcryptMsg_SRT_ParseMsg
};

const hcrypt_MsgInfo* hcryptMsg_SRT_InitMsgInfo(void)
{
	return (&_hcMsg_SRT_MsgInfo);
}
```

**Review by ethouris — APPROVED** [review]

**Pull request merged**